platforms/core-configuration/model-reflect/src/main/java/org/gradle/internal/reflect/annotations/impl/DefaultTypeAnnotationMetadataStore.java

```diff
@@ -526,6 +532,14 @@ private void processPropertyMethodAnnotations(Method method, Map<String, Propert
         if (accessorType == PropertyAccessorType.SETTER) {
             validateNotAnnotatedForProperty(MethodKind.SETTER, method, annotations.keySet(), validationContext);
             validateSetterForMutableType(method, accessorType, validationContext, propertyName);
+            Class<?> propertyType = method.getParameterTypes()[0];
+            String getterName = propertyType.equals(boolean.class)
+                ? "is" + StringUtils.capitalize(propertyName)
+                : "get" + StringUtils.capitalize(propertyName);
+            PropertyAnnotationMetadataBuilder metadataBuilder = propertyBuilders.get(getterName);
+            if (metadataBuilder != null) {
+                validatePropertyAccessorDoesNotOverrideForbiddenToOverrideMethod(method, metadataBuilder);
+            }
```

Comment:
💭 I really don't like how we need to figure out the getter's name from the property name here, but the `propertyBuilders` map is indexed by getter name (to handle is/get getters). I wish there was a better solution, but I don't see one without a significant refactor of the whole class (which we _should_ do at some point).